tls/s2n_signature_algorithms.c

```diff
@@ -104,36 +104,46 @@ int s2n_get_and_validate_negotiated_signature_scheme(struct s2n_connection *conn
     S2N_ERROR(S2N_ERR_INVALID_SIGNATURE_SCHEME);
 }
 
-int s2n_choose_sig_scheme_from_peer_preference_list(struct s2n_connection *conn, struct s2n_sig_scheme_list *peer_wire_prefs,
-                                                        struct s2n_signature_scheme *sig_scheme_out)
+int s2n_choose_default_sig_scheme(struct s2n_connection *conn, struct s2n_signature_scheme *sig_scheme_out)
 {
-    /* This function could be called in two places: after receiving the
-     * ClientHello and parsing the extensions and cipher suites, and after
-     * receiving the CertificateRequest which includes supported signature/hash
-     * pairs. In both cases, the cipher suite will have been set */
+    notnull_check(conn);
+    notnull_check(conn->secure.cipher_suite);
+    notnull_check(sig_scheme_out->sig_alg);
```

Comment:
ok, so clearly we arne't paying close enough attention to each other 🤣 

i meant:

```
notnull_checl(sig_scheme_out)
```